compiler/rustc_passes/src/check_attr.rs

```diff
@@ -250,13 +250,33 @@ impl CheckAttrVisitor<'tcx> {
                                     None
                                 }
                             }
+                            Target::AssocConst => {
+                                let parent_hir_id = self.tcx.hir().get_parent_item(hir_id);
+                                let containing_item = self.tcx.hir().expect_item(parent_hir_id);
+                                // We can't link to trait impl's consts.
+                                let err = "associated constant in trait implementation block";
+                                match containing_item.kind {
+                                    ItemKind::Impl { of_trait: Some(_), .. } => Some(err),
+                                    ItemKind::Impl { ref self_ty, .. } => {
+                                        match self.tcx.hir().find(self_ty.hir_id) {
+                                            Some(Node::Ty(t))
+                                                if matches!(t.kind, TyKind::TraitObject(_, _)) =>
```

Comment:
Why are impls on trait objects treated differently here?

Reply:
Because I didn't want them to be allowed specifically on trait objects. However it can be removed now I guess.